code/modules/mining/equipment_locker.dm

```diff
@@ -557,9 +557,18 @@
 
 /obj/item/weapon/resonator/attack_self(mob/user)
 	if(burst_time == 50)
+		burst_time = 10
+		user << "<span class='info'>You set the resonator's fields to detonate after 1 second.</span>"
+	if(burst_time == 10)
```

Comment:
you set the burst_time to 10, and then you check if the burst_time is 10, I wonder what's going to happen.